pkg/aws/cfnhelper.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+//CompleteStatuses contains all CloudFormation status strings considered complete
+var CompleteStatuses = []string{
+	cloudformation.StackStatusCreateComplete,
+	cloudformation.StackStatusUpdateComplete,
+	cloudformation.StackStatusDeleteComplete,
+}
+
+//FailedStatuses contains all CloudFormation status strings considered failed
+var FailedStatuses = []string{
+	cloudformation.StackStatusCreateFailed,
+	cloudformation.StackStatusRollbackComplete,
+	cloudformation.StackStatusRollbackFailed,
+	cloudformation.StackStatusUpdateRollbackFailed,
+	cloudformation.StackStatusUpdateRollbackComplete,
+	cloudformation.StackStatusDeleteFailed,
+}
+
+//PendingStatuses contains all CloudFormation status strings consideredpending
+var PendingStatuses = []string{
+	cloudformation.StackStatusCreateInProgress,
+	cloudformation.StackStatusDeleteInProgress,
+	cloudformation.StackStatusRollbackInProgress,
+	cloudformation.StackStatusUpdateCompleteCleanupInProgress,
+	cloudformation.StackStatusUpdateInProgress,
+	cloudformation.StackStatusUpdateRollbackCompleteCleanupInProgress,
+	cloudformation.StackStatusUpdateRollbackInProgress,
+	cloudformation.StackStatusReviewInProgress,
+}
+
+// IsFailed tests if the specified string is considered a failed cloudformation stack state
+func IsFailed(status string) bool {
+	for _, s := range FailedStatuses {
+		if s == status {
+			return true
+		}
+	}
+	return false
+}
+
+// IsComplete tests if the specified string is considered a completed cloudformation stack state
+func IsComplete(status string) bool {
+	for _, s := range CompleteStatuses {
+		if s == status {
+			return true
+		}
+	}
+	return false
+}
+
+// IsPending tests if the specified string is considered a pending cloudformation stack state
+func IsPending(status string) bool {
+	for _, s := range PendingStatuses {
+		if s == status {
+			return true
+		}
+	}
+	return false
+}
+
+//StackDoesNotExist Checks if the error recieved for DescribeStacks denotes if the stack is non exsistent
+func StackDoesNotExist(err error) bool {
```

Comment:
I agree with that for the others, like is pending etc.. because the state of the stack "is pending". This one is just nonexistent, so something like "IsMissing" or "DoesNotExist" felt more natural.